eventfiles/class_tests/event_file_views_tests.py

```diff
@@ -37,13 +54,77 @@ def test_render_files_template(self, client, sign_in):
+        char_content = response.content.decode(response.charset)
+        all_file_in_event1 = EventFile.objects.get_files_by_event(EVENT_ID)
+        for file in all_file_in_event1:
+            assert f'{file}' in char_content
+            assert f'{DOWNLOAD_FILE_FROM_EVENT_URL}{file}' in char_content
+            response = client.get('/file/event_file/download/', {'event': EVENT_ID, 'file_name': f'{file}'})
+            assert response.status_code == 200
+            downloaded_file = self.get_file_name_from_response(response)
+            assert f'{file}' in downloaded_file
+
+    def test_delete_button_visible_for_file_user_created(self, sign_in_with_user2, client):
+        """ test that an ordinary user can delete only files he created """
+        char_content = self.get_char_content_from_eventfile_html(client)
+        found_file1, found_file3 = self.find_testfile1_and_testfile3(char_content)
+        assert found_file3 and not found_file1
+
+    def test_unauthorized_user_delete(self, sign_in_with_user2, client):
+        response = client.get(DELETE_EVENT_CREATOR_FILE_URL)
+        assert response.status_code == 401
+
+    def test_delete_file_created_by_user(self, sign_in_with_user2, client):
+        client.get(DELETE_USER2_FILE_URL)
+        assert not EventFile.objects.all().filter(id=FILE_CREATED_BY_USER2).exists()
+
+    def test_all_delete_buttons_visible_for_event_creator(self, sign_in_with_creator, client):
+        """ test that the event creator can delete all files """
+        char_content = self.get_char_content_from_eventfile_html(client)
+        found_file1, found_file3 = self.find_testfile1_and_testfile3(char_content)
+        assert found_file3 and found_file1
+
+    def test_event_creator_delete_any_file(self, sign_in_with_creator, client):
+        client.get(DELETE_USER2_FILE_URL)
+        assert not EventFile.objects.all().filter(id=FILE_CREATED_BY_USER2).exists()
+
+    def test_unsigned_user_delete(self, client):
+        response = client.get(DELETE_USER2_FILE_URL)
+        assert response.status_code == 302
+        response = client.get(response.url)
+        assert response.status_code == 200
+        assertTemplateUsed(response, 'login/register_login.html')
+
+    def test_redirection_to_file_page_after_delete(self, sign_in_with_user2, client):
+        response = client.get(DELETE_USER2_FILE_URL)
+        assert response.status_code == 302
+        response = client.get(response.url)
+        assert response.status_code == 200
+        assertTemplateUsed(response, FILE_CREATION_HTML_PATH)
+
+    def get_file_name_from_response(self, response):
+        header = response.headers
+        return header['Content-Disposition']
+
+    def get_char_content_from_eventfile_html(self, client):
+        response = client.get(reverse("event_files", args=[EVENT_ID]))
+        char_content = response.content.decode(response.charset)
+        return char_content
```

Comment:
this is the same as:
```suggestion
        return response.content.decode(response.charset)
```

Reply:
Done, Thanks 😄